### DISCUSSION TITLE
Put one blank file with java

### DISCUSSION THREAD

**Discussion by xixici** [minio/minio#18102] [Q&A] [answered]
I want to transmit some files from one minio to the another. 
First, I use the api named copyObject to achive it. But, I found it can't useful. The copyObject can't copy data througt different minIO.
So, I just use the native method: getObject and the put it in another minio with PutObject. It is also not useful.

These are my code:
```
 try (InputStream stream =
                                     sourceMinio.getObject(
                                             GetObjectArgs.builder()
                                                     .bucket(sourceParams.getBucketName())
                                                     .object(objectName)
                                                     .build()
                                     )) {
                        targetMinio.putObject(PutObjectArgs.builder().bucket(targetParams.getBucketName()).object(objectName).stream(stream, stream.available(), -1).build());
```

**Comment by xixici** [comment]
Is any another method to solve it?

**Comment by balamurugana** [comment]
Between two different MinIO setup, you have to do `getObject()` from source setup and `putObject()` to destination setup. There is no other way.

**Reply by xixici** [reply]
Yes. It's my method now. But, I found the file is blank with 0 byte. Is anything wrong in my codes? I have no idea about this situation.

**Reply by balamurugana** [reply, marked as answer]
Read https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#available--

**Reply by xixici** [reply]
Thanks a lot. 
The `available()` is not an inteligence method.
 I have read the orgin code about PutObjectArgs and find somthing:
```
Sets stream to upload. Two ways to provide object/part sizes.
If object size is unknown, pass -1 to objectSize and pass valid partSize.
If object size is known, pass -1 to partSize for auto detect; else pass valid partSize to control memory usage and no. of parts in upload.
If partSize is greater than objectSize, objectSize is used as partSize.
A valid part size is between 5MiB to 5GiB (both limits inclusive).
```
So I sovole it by changing my code like
```
targetMinio.putObject(PutObjectArgs.builder().bucket(targetParams.getBucketName()).object(objectName).stream(minioObject, -1, 1024 * 1024 * 5).build());
```
